docs/en/02_Developer_Guides/00_Model/10_Versioning.md

```diff
@@ -318,6 +329,47 @@ The ownership relationship is tracked through an `[image]` [shortcode](/develope
+
+Records can be added to a changeset in the CMS by using the "Add to campaign" button
+that is available on the edit forms of all pages and files. Programmatically, this is done by creating a `SilverStripe\Versioned\ChangeSet` object and invoking its `addObject(DataObject $record)` method.
+
+<div class="info" markdown="1">
+Any DataObject can exist in any number of changesets, and even added to a changeset in advance of being published. While a record need not have modifications to be part of a changeset, for practical purposes, changesets are only concerned with records that have modifications.
+</div>
+
+By including an object in a changeset, the user is implicitly including all of that object's owned records (as declared by the `$owns` setting). This ensures that when a changeset is published (or reverted), the action cascades through not only all of the items explicitly added to the changeset, but also all of the records that each of those items owns.
+
+#### ChangeSet actions
+
+Actions available on the frontend, i.e. those which are intended to be triggered by an end user, include:
+
+* `$myChangeSet->addObject(DataObject $record)`: Add a record and all of its owned records to the changeset (`canEdit()` dependent)
+* `$myChangeSet->removeObject(DataObject $record)`: Removes a record and all of its owned records from the changeset (`canEdit()` dependent)
+* `$myChangeSet->::publish()`: Publishes all items in the changeset that have modifications, along with all their owned records (`canPublish()` dependent). Closes the changeset on completion.
+* `$myChangeSet->::revert()`: Reverts all items in the changeset that have modifications, along with all their owned records (`canRevert()` dependent). Closes the changeset on completion.
+ 
+ <div class="info" markdown="1">
+ Changesets cannot be "unpublished." Instead, a new changeset should be created to reverse the actions of the prior one.
+ </div>
+ 
+ Actions available on the backend, i.e. those which should only be invoked programmatically include:
+ 
+ * `$myChangeSet->sync()`: Find all owned records with modifications for each item in the changeset, and include them implicitly.
+ * `$myChangeSet->validate()`: Ensure all owned records with modifications for each item in the changeset are included. This method should not need to be invoked if `sync()` is being used on each mutation to the changeset.
+ 
+ #### ChangeSet states
+ 
+ Changesets can exists in five different states:
+ 
+* `open` No action has been taked on the changeset. Resolves to `publishing` or `reverting`.
+* `publishing`:  The changeset is in the process of publishing its items. Resolves to to `published`.
+* `reverting`: The changeset is in the process of reverting changes to its items. Resolves to `reverted`.
+* `published`: The changeset has published changes to all of its items and its now closed.
+* `reverted`: The changeset has reverted changes to all of its items and its now closed.
+ 
 ### Custom SQL
```

Comment:
We should also document the changetype, and describe what each of them means in the campaigns view. I.e.:

- created
- modified
- deleted
- none